easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java

```diff
@@ -125,11 +125,13 @@ public static void requestPermissions(@NonNull final Object object, @NonNull Str
      *         request code to track this request, must be < 256.
      * @param perms
      *         a set of permissions to be requested.
+     * @return true if a permissions request was started, false otherwise.
      */
     @SuppressLint("NewApi")
-    public static void requestPermissions(@NonNull final Object object, @NonNull String rationale,
+    public static boolean requestPermissions(@NonNull final Object object, @NonNull String rationale,
```

Comment:
Although I see both sides, I like the explicitness of the current model where you say if(!hasPermissions()) { requestPermissions() } in your app code since then you know exactly when you're about to launch UI.  I think it also matches the simplest mental model.